frontend/app/assets/javascripts/linker.js

```diff
@@ -284,6 +284,24 @@ $(function() {
 
       var tokensForPrepopulation = function() {
         if ($this.data("multiplicity") === "one") {
+
+          // If we are on a resource edit page, and open a top_container modal with a collection_resource linker
+          // then we prepopulate the collection_resource field with resource data necessary to perform the search
+          let currentPath = window.location.pathname;
+          if (currentPath.match(/^\/resources\/[\s\S]*\/edit$/) && $(".label.label-info").text() === "Resource" && $(".modal-dialog").find("#collection_resource").length > 0 && $this[0].id === "collection_resource") {
```

Comment:
This regex should made more specific - `[\s\S]it reads "nonspace or space", which matches '/' - the fixed `/edit$/` at the end allows this to work, but that token is eating the entire rest of the string and then giving up one character at a time until it can match the fixed string, a process called backtracking, which is bad for performance (in general, not gonna matter over this short a run of chars) and introduces fragility (if this got altered to apply to the same page minus the fixed ending, it would break).

My recommendation is to change the regex with an eye toward making it as restrictive as possible while matching safely.  That parameter is only and exclusively resource IDs, which I am 99% are reliably integers (I don't know of any plugins that make ASpace use GUIDs or something, anyway). 

The full list of criteria as I see them are:

- integer (\d)
- at least one (+ quantifier rather than *)
- You genuinely want all digits no matter how many, so it makes sense to leave the quantifier greedy

So, suggested regex: `/^\/resources\/\d+\/edit$/`